pkg/eks/eks.go

```diff
@@ -184,7 +184,7 @@ func (c *ClusterProvider) maybeRefreshClusterStatus(spec *api.ClusterConfig) err
 func (c *ClusterProvider) CanDelete(spec *api.ClusterConfig) (bool, error) {
 	err := c.maybeRefreshClusterStatus(spec)
 	if err != nil {
-		if awsError, ok := errors.Unwrap(errors.Unwrap(err)).(awserr.Error); ok &&
+		if awsError, ok := err.(awserr.Error); ok &&
```

Comment:
I _think_ `https://golang.org/pkg/errors/#As` might fit here and then it can be left wrapped above?

Reply:
Thats neat! I do sort of unlike the idea that the caller of `DescribeCluster` has to know to unwrap the error, but saying that perhaps its wrong that the caller is relying on it returning an awseks error 🤔  

My brain hurts. I'm going to make the change you suggested

Reply:
I'm not sure I've ever used `errors.As`  so no guarantees

Reply:
I tried it out and it worked 👍

Reply:
actually I don't think it worked, I'm not sure why exactly... I'm playing around with it to see if I can get it to work as its much cleaner

Reply:
I played around and couldn't get it to work 🤷  just reverted back to the unwrap approach.